### PR TITLE
feat(athena): add amazon athena backend

### PR DESCRIPTION
**Pull request by cpcloud**
Draft PR for the Amazon Athena backend. Waiting on Voltron Data internal help with CI setup.

### CODE REVIEW

**Comment by cpcloud** [comment]
All passing locally:

```
cloud in 🌐 falcon in …/ibis on  athena is 📦 v9.5.0 via 🐍 v3.12.8 via ❄️  impure (ibis-3.12-env) took 32s
❯ pytest -m athena -n 20 --snapshot-update -q
bringing up nodes...
.ssssssssssssssssssssssssx......ssssssssssssssssssssssss......ssssxxxxx.x.xx.........x....x.........x......... [  5%]
.........xx.........xssssssssssssssssssssssss..........x.....xx..............x...x.x.......xssssssssssssssssss [ 11%]
sssss....xxxx......x.x..........................x..x.x.s.x.x.x.........x...................................... [ 17%]
..xx..x...x.x......xx......x...............x................xxx...x...x.....x....x..x.......xxxxxxxxxx.......x [ 22%]
.......x.......x.........x...........x.....xx.xx.....x...x..............................x..........x.x....xx.. [ 28%]
.x.......x...x....................x...................x..............x...xxx............x............x........ [ 34%]
.....x.......x.x....x........x......x..................x..x.................X.................x...x........x.. [ 39%]
.....................x.....x....................x.............x..........Xx.....................x............. [ 45%]
.......x...x.....x..................xxxx........x....x..xx..............x........x.......x.................... [ 51%]
..x..........x....xxxxxxxxxx...xxxxxxx..x.........sx..................x..x.x.x......x.....x..xxx.............. [ 56%]
.x...x....x....x.......x..x.......x......x...x....s.........x.....x.x..........xsx....xx...x.....x...x...xxx.. [ 62%]
x.x..x.s..ssssssssssssss...........x..x...xx.........xs.........xxx......sssssss....x.x....................... [ 68%]
..x......xx.x.x.s..xx.x..x..x.........x..........xx...x.....x...............x..x......xx....x......xx.xx...x.. [ 73%]
..x.x...x....x..x..xxxxxxxxx.x.....x.x..xx..............x.....xxxxxx...x.x....x.x.x.x.x.....xx....xxx..s.x.x.x [ 79%]
..x...........x..x.xx.x..x.xx.x.x......x..x..x....x.xx...xx.xxxx.x.xx.xx..x.xxxx....x.x.x.x...x........x..xx.. [ 85%]
x.x.........xx.xx...xx.x.x..xxx....xxx.xx..xxx........................................s.s.....x............... [ 91%]
.......................................................xx.................................................x... [ 96%]
....xx.x.x...xx.....x......xxx...x.x.....x..x.....x..x.x.xxx..x.                                               [100%]
1464 passed, 130 skipped, 338 xfailed, 2 xpassed in 546.92s (0:09:06)
```

File: docs/backends/athena.qmd [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+### Connection Parameters
+
+```{python}
+#| echo: false
+#| output: asis
+from _utils import render_do_connect
+
+render_do_connect("athena")
+```
+
```
Comment (gforsyth) [inline docs/backends/athena.qmd:111]:
```{python}
#| echo: false
BACKEND = "BigQuery"
```

Should add this in here for the docs scripts

**Review by gforsyth — COMMENTED** [review]
Looks reasonable -- love a cloud-hosted older version of an existing backend with its own quirks.

Only blocker right now is getting the tests running in CI

File: docs/backends/athena.qmd (hunk 1 above)
Comment (cpcloud) [reply docs/backends/athena.qmd:111]: Done! Thank you!

**Comment by IndexSeek** [comment]
It's always cool to see a new backend! Is it worth adding a note to the [cloud_support_policy](https://github.com/ibis-project/ibis/blob/main/docs/backends/cloud_support_policy.qmd)?

**Comment by cpcloud** [comment]
> It's always cool to see a new backend! Is it worth adding a note to the [cloud_support_policy](https://github.com/ibis-project/ibis/blob/main/docs/backends/cloud_support_policy.qmd?rgh-link-date=2024-12-31T13%3A18%3A01Z)?

Absolutely, good catch!

**Comment by cpcloud** [comment]
Ok, I was able to set up the AWS credentials. I can debug that once it's merged.

**Pull request merged**